lib/tbd/psi.rb

```diff
@@ -268,110 +419,141 @@ def gen(id = "")
+      v[:sill              ] = @set[id][:fenestration      ] if h[:fenestration      ]
+      v[:sillconcave       ] = @set[id][:fenestration      ] if h[:fenestration      ]
+      v[:sillconvex        ] = @set[id][:fenestration      ] if h[:fenestration      ]
+      v[:jamb              ] = @set[id][:fenestration      ] if h[:fenestration      ]
+      v[:jambconcave       ] = @set[id][:fenestration      ] if h[:fenestration      ]
+      v[:jambconvex        ] = @set[id][:fenestration      ] if h[:fenestration      ]
+      v[:head              ] = @set[id][:head              ] if h[:head              ]
+      v[:headconcave       ] = @set[id][:head              ] if h[:head              ]
+      v[:headconvex        ] = @set[id][:head              ] if h[:head              ]
+      v[:sill              ] = @set[id][:sill              ] if h[:sill              ]
+      v[:sillconcave       ] = @set[id][:sill              ] if h[:sill              ]
+      v[:sillconvex        ] = @set[id][:sill              ] if h[:sill              ]
+      v[:jamb              ] = @set[id][:jamb              ] if h[:jamb              ]
+      v[:jambconcave       ] = @set[id][:jamb              ] if h[:jamb              ]
+      v[:jambconvex        ] = @set[id][:jamb              ] if h[:jamb              ]
+      v[:headconcave       ] = @set[id][:headconcave       ] if h[:headconcave       ]
+      v[:headconvex        ] = @set[id][:headconvex        ] if h[:headconvex        ]
+      v[:sillconcave       ] = @set[id][:sillconcave       ] if h[:sillconcave       ]
+      v[:sillconvex        ] = @set[id][:sillconvex        ] if h[:sillconvex        ]
+      v[:jambconcave       ] = @set[id][:jambconcave       ] if h[:jambconcave       ]
+      v[:jambconvex        ] = @set[id][:jambconvex        ] if h[:jambconvex        ]
+      v[:corner            ] = @set[id][:corner            ] if h[:corner            ]
+      v[:cornerconcave     ] = @set[id][:corner            ] if h[:corner            ]
+      v[:cornerconvex      ] = @set[id][:corner            ] if h[:corner            ]
+      v[:cornerconcave     ] = @set[id][:cornerconcave     ] if h[:cornerconcave     ]
+      v[:cornerconvex      ] = @set[id][:cornerconvex      ] if h[:cornerconvex      ]
+      v[:parapet           ] = @set[id][:parapet           ] if h[:parapet           ]
+      v[:parapetconcave    ] = @set[id][:parapet           ] if h[:parapet           ]
+      v[:parapetconvex     ] = @set[id][:parapet           ] if h[:parapet           ]
+      v[:parapetconcave    ] = @set[id][:parapetconcave    ] if h[:parapetconcave    ]
+      v[:parapetconvex     ] = @set[id][:parapetconvex     ] if h[:parapetconvex     ]
+      v[:roof              ] = @set[id][:parapet           ] if h[:parapet           ]
+      v[:roofconcave       ] = @set[id][:parapet           ] if h[:parapet           ]
+      v[:roofconvex        ] = @set[id][:parapet           ] if h[:parapet           ]
```

Comment:
If a PSI set does not hold a roof (not parapet) type, TBD recovers the PSI factor of the parapet type as back-up.